pageserver/src/page_service.rs

```diff
@@ -2236,9 +2236,10 @@ impl PageServerHandler {
 
         let ctx = match perf_instrument {
             true => RequestContextBuilder::from(ctx)
-                .root_perf_span(|| {
+                .perf_span(|get_page_span| {
```

Comment:
Each get page request read from the network has its own context in `BatchedGetPageRequest`. That's the context that will get a root perf span if the request is sampled.

Here you're taking whatever the current perf span is and making `GET_VECTORED` a child of it.
It's unexpected that `ctx` is perf enabled. This should be a child of `PageServerHandler::connection_ctx`
which should never have a perf span.

Could you share your pageserver config?

Reply:
More generally, we can't make the `GET_VECTORED` span a child of `GET_PAGE` since the vectored get serves all the requests in the batch, so it would need to have multiple parents. This isn't supported by OTEL, so we use a follows-from relationship instead.

Reply:
What you said makes sense. I also wasn't sure if this is the right.

This is my toml config:

```
...
tracing = { sampling_ratio = { numerator = 100, denominator = 100 }, export_config = { endpoint = 'otel_http_endpoint', protocol = 'http-binary', timeout = '5s'} }
...
```

Reply:
Sure! This is the full config:

```
broker_endpoint = 'http://foobar'
pg_distrib_dir = '/usr/local/'
listen_pg_addr = '0.0.0.0:6400'
listen_http_addr = '0.0.0.0:9898'
disk_usage_based_eviction = { max_usage_pct = 90, min_avail_bytes = 2154487209164, period = '1m' }
control_plane_emergency_mode = true
remote_storage = { endpoint = 'barfoo', bucket_name = 'bucketname', bucket_region='region', prefix_in_bucket = '/pageserver' }
tenant_config = { pitr_interval = '1 day' }
tracing = { sampling_ratio = { numerator = 100, denominator = 100 }, export_config = { endpoint = 'http://otel-collector-otlp-http-chi1.simulprod.com', protocol = 'http-binary', timeout = '5s'} }
```

Reply:
hmm, somehow, I am not able to repro this with our environment. Very strange.

Reply:
Oh, another thing I have just noticed is that I was using this docker image: `neondatabase/neon:7789`. Not sure what was the diff but when I am using `neondatabase/neon:7897`, this panic doesn't happen anymore.